tools/FileTestDefs.sh

```diff
@@ -581,6 +581,46 @@ prevent_cpp_includes_test() {
 }
 standard_checks+=(prevent_cpp_includes)
 
+# Check for CMakeLists.txt hardcoding library names
+cmakelists_hardcoded_libraries_command_list() {
+    printf '%s' add_dependencies
+    printf '|%s' add_test_library spectre_python_add_dependencies \
+           spectre_python_link_libraries spectre_target_headers \
+           spectre_target_sources target_link_libraries target_link_options
+}
+cmakelists_hardcoded_libraries() {
+    local CHECKED_COMMANDS=$(cmakelists_hardcoded_libraries_command_list)
+    [[ $1 =~ CMakeLists\.txt$ ]] && \
+        whitelist "$1" \
+                  "tests/Unit/Parallel/CMakeLists.txt$" \
+                  "tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/\
+CMakeLists.txt$" && \
+        staged_grep -E -A1 "(${CHECKED_COMMANDS})\(\$" "$1" | \
+            grep -Ev -- "--|${CHECKED_COMMANDS}" | \
+            grep -qvF '${'
+}
+cmakelists_hardcoded_libraries_report() {
+    local CHECKED_COMMANDS=$(cmakelists_hardcoded_libraries_command_list)
+    echo "Found CMakeLists.txt files using hardcoded library names."
```

Comment:
Could you add a brief instruction what the person encountering this error should do instead, or link to a place that describes it?

Reply:
ping @nilsleiffischer

Reply:
I suppose that works. I think it would also be a good idea to update https://spectre-code.org/spectre_build_system.html as well.

Reply:
That page already says to use a ${LIBRARY} variable.  What should I update?

Reply:
It could specify what to do in subdirectories that don't define a new library. That's the main issue this PR addresses if I understand correctly. I'm mostly just worried that defining the variable in the parent scope and using it in the subdirectory can be confusing.

Reply:
Oh, no, that's not the point of this at all.  That was just an easier change for me than adding the additional lines to redefine the variable in all the files I had to touch.  I like this way of doing it, but I'm not proposing it as a standard or anything.

The point of this was catching cases where inconsistent values were used, like this case from #2966
```
set(LIBRARY VariableFixing)

spectre_target_sources(
  ${LIBRARY}
  ...
  )

spectre_target_headers(
  Evolution
  ...
  )
```

Reply:
I see. How about you add to the error message: "Use a variable to ensure the library name is consistent within the file" or something along those lines.

Reply:
Yes, that's a good message.  Thanks.